src/gax-internal/src/http.rs

```diff
@@ -371,33 +372,33 @@ mod test {
         Ok(())
     }
 
-    macro_rules! test_client_empty_content {
-        ($name:ident, $($code:expr, $content:expr, should_fail $should_fail:expr),+) => {
-            #[tokio::test]
-            async fn $name() -> TestResult {
-                let http_resp = http::Response::builder()
-                    .header("Content-Type", "application/json")
-                    .status($($code)*)
-                    .body($($content)*)?;
-                let response: reqwest::Response = http_resp.into();
-                assert!(response.status().is_success());
-
-                let response = ReqwestClient::to_http_response::<wkt::Empty>(response).await;
-                if $($should_fail)* {
-                    assert!(response.is_err());
-                    return Ok(());
-                }
-                assert!(response.is_ok());
+    #[tokio::test]
+    #[test_case(reqwest::StatusCode::OK, "{}", false; "200 with empty object")]
+    #[test_case(reqwest::StatusCode::NO_CONTENT, "{}", false; "204 with empty object")]
+    #[test_case(reqwest::StatusCode::OK, "", true; "200 with empty content")]
+    #[test_case(reqwest::StatusCode::NO_CONTENT, "", false; "204 with empty content")]
+    async fn client_empty_content(
+        code: reqwest::StatusCode,
+        content: &str,
+        should_fail: bool,
```

Comment:
optional: go/tott/661

Consider splitting this into a test function for the `should_fail == true` base and  one for the `should_fail == false` case.  I know it is also bad to refactor test code, but a function to create the `response` would make those two functions pretty small.

Reply:
done